WcaOnRails/app/models/user.rb

```diff
@@ -293,6 +285,44 @@ def after_confirmation
     end
   end
 
+  def software_team?
+    team_member?('software') != false
+  end
+
+  def results_team?
+    team_member?('results') != false
+  end
+
+  def wrc_team?
+    team_member?('wrc') != false
+  end
+
+  def wdc_team?
+    team_member?('wdc') != false
+  end
+
+  def team_member?(team)
+    member = self.team_members.find_by_team_id( self.teams.find_by_friendly_id(team) )
+    if member && (member.end_date == nil || member.end_date >= Date.today)
```

Comment:
I actually thought of this the other day, but didn't remember to check the
code.

Short answer: it doesn't work in the case you described :P

I'll look into it.

On Mon, Mar 21, 2016 at 4:55 AM, Jeremy Fleischman <notifications@github.com
> wrote:

> In WcaOnRails/app/models/user.rb
> <https://github.com/cubing/worldcubeassociation.org/pull/390#discussion_r56789108>
> :
>
> > +
> > +  def results_team?
> > +    team_member?('results') != false
> > +  end
> > +
> > +  def wrc_team?
> > +    team_member?('wrc') != false
> > +  end
> > +
> > +  def wdc_team?
> > +    team_member?('wdc') != false
> > +  end
> > +
> > +  def team_member?(team)
> > +    member = self.team_members.find_by_team_id( self.teams.find_by_friendly_id(team) )
> > +    if member && (member.end_date == nil || member.end_date >= Date.today)
>
> This looks like another place where having a method like current_member?
> on team_members would be nice.
>
> Can you explain how this works with people who were on and off teams? It
> looks like you're only looking at one of their team_members, rather than
> all of them.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/cubing/worldcubeassociation.org/pull/390/files/4cfc692de289d2d4d24e03be7e8db4b02da2a89d#r56789108>
>